### ISSUE TITLE
Usage of `objc_enable_binary_stripping` results in linker errors

### ISSUE THREAD

**Issue by rahul-malik** [bazelbuild/bazel#6167]
### Description of the problem / feature request:

Using `objc_enable_binary_stripping` feature will pass the necessary `dead_strip` flags to clang during the linking phase. The current OSX CROSSTOOL does not pass the correct linker flags.

I've put up a patch to address this here https://github.com/bazelbuild/bazel/pull/6166

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run `bazel build` on a ObjC target that uses clang / ld with `--objc_enable_binary_stripping=true` specified 

### What operating system are you running Bazel on?

macOS High Sierra

### What's the output of `bazel info release`?

release 0.16.1

###  Have you found anything relevant by searching the web?
No

### Any other information, logs, or outputs that you want to share?
Current errors without a patch in the CROSSTOOL
```
clang++: error: unsupported option '--dead_strip'
clang++: error: unsupported option '--no_dead_strip_inits_and_terms'
```

**Comment by aragos** [comment]
Fixed in c361da0

**Comment by rahul-malik** [comment]
@aragos - Oops sorry I didn't close this after merging
